packages/pigeon/lib/dart_generator.dart

```diff
@@ -74,6 +76,39 @@ void generateDart(Root root, StringSink sink) {
         }
       });
       indent.writeln('');
+    } else if (api.location == ApiLocation.flutter) {
```

Comment:
subjective style nit: At this point I'd  this point this function should really be split up into private helper methods. It's really long, multiple levels of scoping and for loops, and relatively complex. I think it's OK to have private helper methods that are just called once, it brings the cyclomatic complexity of whatever the reader has to understand at one point down.

Reply:
done, split off some helper functions